File: challenges/3-tipos-de-triangulo/python/virusxdrj/tipos_de_triangulos.py
from sys import argv
from dataclasses import dataclass


@dataclass
class Triangle:
    a: float
    b: float
    c: float 


def type_of_triangle(sides: Triangle) -> None:
    if sides.a >= sides.b + sides.c:
        print("NAO FORMA TRIANGULO")
        return
    if sides.a**2 == (sides.b**2 + sides.c**2):
        print("TRIANGULO RETANGULO")
    if sides.a**2 > (sides.b**2 + sides.c**2):
        print("TRIANGULO OBTUSANGULO")
    if sides.a**2 < (sides.b**2 + sides.c**2):
        print("TRIANGULO ACUTANGULO")
    if sides.a == sides.b == sides.c:
        print("TRIANGULO EQUILATERO")
        return
    if sides.b == sides.c or sides.a == sides.b:
        print("TRIANGULO ISOSCELES")


if __name__ == "__main__":
    args: list[float] = [float(arg) for arg in argv[1:4]]
    args: list[float] = sorted(args, reverse=True)
    type_of_triangle(Triangle(a=args[0], b=args[1], c=args[2]))

File: challenges/3-tipos-de-triangulo/javascript/tozeos/tiposdetriangulos.js
// TIPOS DE TRIÂNGULOS

const arg1 = Number(process.argv.slice(2, 3));
const arg2 = Number(process.argv.slice(3, 4));
const arg3 = Number(process.argv.slice(4, 5));

var lado1 = 0;
var lado2 = 0;
var lado3 = 0;

// Verificação a organização de lados.

if (arg1 <= arg2 && arg1 <= arg3) {
	lado3 = arg1;
	if (arg2 <= arg3) {
		lado2 = arg2;
		lado1 = arg3;
	} else {
		lado2 = arg3;
		lado1 = arg2;
	}
} else if (arg2 <= arg1 && arg2 <= arg3) {
	lado3 = arg2;
	if (arg1 <= arg3) {
		lado2 = arg1;
		lado1 = arg3;
	} else {
		lado2 = arg3;
		lado1 = arg1;
	}
} else if (arg3 < arg1) {
	lado3 = arg3;
	if (arg1 < arg2) {
		lado2 = arg1;
		lado1 = arg2;
	} else {
		lado2 = arg2;
		lado1 = arg1;
	}
}

// Classificação

if (lado1 >= lado2 + lado3) {
	console.log("NAO FORMA TRIANGULO");
} else {
	if (Math.pow(lado1, 2) == Math.pow(lado2, 2) + Math.pow(lado3, 2)) {
		console.log("TRIANGULO RETANGULO");
	}

	if (Math.pow(lado1, 2) > Math.pow(lado2, 2) + Math.pow(lado3, 2)) {
		console.log("TRIANGULO OBTUSANGULO");
	}

	if (Math.pow(lado1, 2) < Math.pow(lado2, 2) + Math.pow(lado3, 2)) {
		console.log("TRIANGULO ACUTANGULO");
	}

	if (lado1 == lado2 && lado1 == lado3) {
		console.log("TRIANGULO EQUILATERO");
	}

	if (
		(lado1 == lado2 && lado1 != lado3) ||
		(lado1 == lado3 && lado1 != lado2) ||
		(lado2 == lado3 && lado2 != lado1)
	) {
		console.log("TRIANGULO ISOSCELES");
	}
}
